archunit/src/test/java/com/tngtech/archunit/testutil/assertion/ArchRuleCheckAssertion.java

```diff
@@ -0,0 +1,67 @@
+package com.tngtech.archunit.testutil.assertion;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Optional;
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.EvaluationResult;
+import org.assertj.core.api.AbstractObjectAssert;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.guava.api.Assertions.assertThat;
+
+public class ArchRuleCheckAssertion extends AbstractObjectAssert<ArchRuleCheckAssertion, ArchRuleCheckAssertion.Check> {
```

Comment:
Does extending `AbstractObjectAssert` give any benefit if the `actual` object is constructed as an internal `Check`?
The user of `ArchRuleCheckAssertion` has no idea about the inner class, and for the implementaition you always have to write two methods for one feature.

Reply:
I think this is because originally it was no internal `Check` :wink: Yeah, you're right, I don't think the current implementation has any benefit of the extension...